event-schema/event_schema_test.go

```diff
@@ -1,13 +1,202 @@
+	var eventPayload EventPayloadT
+	if err := json.Unmarshal([]byte(eventStr), &eventPayload); err != nil {
+		t.Errorf("Invalid request payload for unmarshalling: %v", err.Error())
+		return
+	}
+
+	// Send event across a new write key
+	manager.handleEvent(writeKey, eventPayload.Batch[0])
+
+	// Event Model gets stored in the in-memory map
+	eventModel := manager.eventModelMap[WriteKey(writeKey)][("track")][("Demo Track")]
+	require.NotEmpty(t, eventModel.UUID)
+	require.Equal(t, eventModel.EventIdentifier, "Demo Track")
+
+	// Schema Version gets stored in the in-memory map for event model.
+	eventMap := map[string]interface{}(eventPayload.Batch[0])
+	flattenedEvent, _ := flatten.Flatten(eventMap, "", flatten.DotStyle)
+	hash := getSchemaHash(getSchema(flattenedEvent))
+
+	schemaVersion := manager.schemaVersionMap[eventModel.UUID][hash]
+	require.NotNil(t, schemaVersion)
+}
+
+func TestHandleEventBoundsFrequencyCounter(t *testing.T) {
+	t.Log("Testing the process to bound frequency counters within event model")
+
+	writeKey := "my-write-key"
+	manager := EventSchemaManagerT{
+		dbHandle:             pgResource.DB,
+		disableInMemoryCache: false,
+		eventModelMap:        EventModelMapT{},
+		schemaVersionMap:     SchemaVersionMapT{},
+	}
+	eventStr := `{"batch": [{"type": "track", "event": "Demo Track", "properties": {"label": "Demo Label", "value": 5 }}], "writeKey": "my-write-key", "requestIP": "127.0.0.1", "receivedAt": "2022-06-15T13:51:08.754+05:30"}`
+	var eventPayload EventPayloadT
+	if err := json.Unmarshal([]byte(eventStr), &eventPayload); err != nil {
+		t.Errorf("Invalid request payload for unmarshalling: %v", err.Error())
+		return
+	}
+
+	manager.handleEvent(writeKey, eventPayload.Batch[0])
+	eventModel := manager.eventModelMap[WriteKey(writeKey)]["track"]["Demo Track"]
+
+	// Push the events to DB
+	manager.flushEventSchemasToDB()
+
+	// Bound the frequency counters to 3
+	frequencyCounterLimit = 3
+
+	// reload the models from the database which should now respect
+	// that frequency counters have now been bounded.
+	manager.handleEvent(writeKey, eventPayload.Batch[0])
+	require.Equal(t, len(countersCache[eventModel.UUID]), 3)
+
+	// flush the events back to the database.
+	manager.flushEventSchemasToDB()
+	freqCounters, err := getFrequencyCountersForEventModel(manager.dbHandle, eventModel.UUID)
+	require.Nil(t, err)
+	require.Len(t, freqCounters, 3)
+}
+
+func getFrequencyCountersForEventModel(handle *sql.DB, uuid string) ([]*FrequencyCounter, error) {
+	query := `SELECT private_data FROM event_models WHERE uuid = $1`
+	var privateDataRaw json.RawMessage
+	err := handle.QueryRow(query, uuid).Scan(&privateDataRaw)
+	if err != nil {
+		return nil, err
+	}
+
+	var privateData PrivateDataT
+	err = json.Unmarshal(privateDataRaw, &privateData)
+	if err != nil {
+		return nil, err
+	}
+
+	return privateData.FrequencyCounters, nil
+}
+
+func BenchmarkEventSchemaHandleEvent(b *testing.B) {
```

Comment:
What has this benchmark revealed? Have you done a before and after so that we can compare results?

Reply:
I can't find the report, have you added it somewhere?